### ISSUE TITLE
Update Order of Operations Lesson

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#1543]
- use @lesson-link to connect to more specific em lessons.
- move pages from order of operations into lessons that could use them
- identify which other lessons need to be written 
- #574
- [ ] ES requests no negatives or exponents in required materials - Should this include?
-- printed cards where all students do is match them
-- Pages where they are completing circles of evaluation and code and don't have to touch the parts with negatives?
-- The 3-column-challenge page we reference inline in the lesson?

**Comment by flannery-denny** [comment]
@retabak @schanzer I know that you both think that this lesson is overwhelming, but there's actually a lot of stuff missing and the pages that made it into the workbook this year are much harder than how this used to start!  I know that I built out all of this complicated stuff, but I don't think that my students would have been successful without the intro materials... I feel like I've missed out on a big discussion about this lesson that must have happened to pair it down? Adding to our curriculum team meeting in hopes that someone can bring me up to speed.

**Comment by flannery-denny** [comment]
Interesting. In that case I'll just resurrect things if they feel necessary. Thanks @retabak

**Comment by flannery-denny** [comment]
@schanzer - Do you feel ok about saving the grammar discussion for simple data types?

Order of Operations

> Humans use verbs like "throw", "run", "build" and "jump" to describe operations on nouns like "ball", "puppy", and "blocks". Mathematics has "operations" , like addition and subtraction. Just as you can "throw a ball", a person can also "add four and five".
> 
> A mathematical expression is an instruction for doing something, which specifies the verbs and nouns involved. The expression 4 + 5 tells us to add 4 and 5. To evaluate an expression, we follow the instructions. The expression 4 + 5 evaluates to 9.

Simple Data Types

> Math is a language, just like English, Spanish, or any other language. Languages have nouns (e.g. “ball”, “tomato”, etc.) and verbs, which are actions we can perform on these nouns (e.g. - I can “throw a ball”). Math and programming also have values, like the numbers 1, 2 and 3. And, instead of verbs, they have functions, which are actions we can perform on values (e.g. - “I can square a number”).
> 
> Languages also have rules for syntax. In English, for example, words don’t have ! and ? in the middle. In math and programming numbers don’t have & in them.
> 
> Languages also have rules for grammar. The cat sat. is a sentence, whereas The sat cat. is nonsense, even though all the words are valid syntax. The order of the words matters!
> 
> Keeping the importance of syntax and grammar in mind is helpful when learning to program!

**Comment by schanzer** [comment]
@flannery-denny definitely.

As long as the contract lessons make a brief reference back, to make sure that kids understand that functions are verbs, too.

**Comment by flannery-denny** [comment]
@ds26gte is there a way for me to use `show{(coe '(- 5 (/ (+ 7 8) 2)))}` inline in a pedagogy box? Or would I need to build it, screenshot it, and save it as an image first? 

I am trying to write

`@left{@show{(coe '(- 5 (/ (+ 7 8) 2)))}}The Circles of Evaluation are a critical pedagogical tool in this course. They place the focus on the _structure_ of mathematical expressions, as a means of combating the harmful student belief that the only thing that matters is the _answer_. They can be used to diagram arithmetic sentences to expose common misconceptions about Order of Operations, and make an excellent scaffold for tracing mistakes when a student applies the Order of Operations incorrectly. They are also a bridge representation, which naturally connects to function composition and converting arithmetic into code.`

But I am getting

<img width="633" alt="Screen Shot 2023-10-13 at 1 36 50 PM" src="https://github.com/bootstrapworld/curriculum/assets/66691020/9a34bbea-ca2b-4ae3-ab5c-2a11a54dcae2">

**Comment by schanzer** [comment]
@flannery-denny if this is on 2024, can you push a commit and I'll take a look. Dorai is MIA for the rest of the day

**Comment by flannery-denny** [comment]
Pushing where my thinking is at (not quite the same as in the screenshot above).
For the record, I also tried using a table within the pedagogy box and it didn't work with `|===` or `!===`

**Comment by flannery-denny** [comment]
@schanzer ps. While you're at it, I removed the following css because it wasn't actually working. If you have other ideas about how to increase the size of **6 ÷ 2(1 + 2)**, I'm all ears.
```
.centered-image.big, .centered-image.big p {margin-top: 0px; padding-top: 0px;}
.big .MathJax {font-size: 4em; color: black;}
```

**Comment by flannery-denny** [comment]
@schanzer 
ps. This is not blocking me on thinking about the lesson. I knew Dorai was offline, and it can wait until he's back.

**Comment by schanzer** [comment]
@flannery-denny I've pushed an update to make the math big. Now that we're not using mathjax, we needed to change the selector at the top of the file.

I've also pushed a larger change, which switches all Circles of Evaluation to be constructed from SPAN elements instead of DIVs. AsciiDoc renders every line of text as a SPAN and spans aren't allowed to contain DIVs, so this was preventing two circles from being on the same line in your strategy box.

@ds26gte do we have any control over AsciiDoctor's use of SPAN elements? This isn't the first time we've run into this, and in the process of trying to fix this I noticed that it _also_ causes problems with TABLE elements. We could have fixed all of these upfront by getting AsciiDoctor to use DIVs instead of SPANs, since a DIV can contain _anything_

**Comment by flannery-denny** [comment]
Thanks @schanzer. I think that most of the time we really don't want multiple coe inline... 

But, for this strategy box I'm still not quite getting what I'm looking for (see screenshot below of what is currently possible with inline text)

My question for you and @ds26gte is two fold:
1. Is there a way to make this coe inline with the text without using a screenshot to make it read as an image as a workaround?
2. Can we adjust the @strategy directive so that I can use tables inside of them?

<img width="646" alt="Screen Shot 2023-10-16 at 11 41 54 AM" src="https://github.com/bootstrapworld/curriculum/assets/66691020/61e4d599-2098-42a4-9220-8b01a9b520bf">

**Comment by flannery-denny** [comment]
@ds26gte also wondering. if QandA could include tables?

```
@QandA{

@Q{What expressions do you think the Circles of Evaluation on  @printable-exercise{coe-notice.adoc}represent?}
@A{

[.embedded, cols="1a,2a,2a,3a,3a", options="header"]
|===
|1|2|3|4|5

|@show{(coe '(* 5 6))} 
|@show{(coe '(* (/ 15 3) 6))}
|@show{(coe '(* 5 (- 28 22)))}
|@show{(coe '(* (/ 15 3) (- 28 22)))}
|@show{(coe '(* (/ 15 (- 4 1)) 2))}

|@math{5 \times 6}}
|@math{15 \div 3 \times 6}
|@math{5 \times (28 - 22)}
|@math{15 \div 3 \times (28 - 22)}
|@math{15 \div (4 - 1) \times 2}
|===

}
```

for now I'm using

```
@lesson-instruction{
* What expressions do you think the Circles of Evaluation on  @printable-exercise{coe-notice.adoc} represent?
}
@teacher{

[cols="1a,2a,2a,3a,3a", options="header"]
|===
|1|2|3|4|5

|@show{(coe '(* 5 6))} 
|@show{(coe '(* (/ 15 3) 6))}
|@show{(coe '(* 5 (- 28 22)))}
|@show{(coe '(* (/ 15 3) (- 28 22)))}
|@show{(coe '(* (/ 15 (- 4 1)) 2))}

|@math{5 \times 6}}
|@math{15 \div 3 \times 6}
|@math{5 \times (28 - 22)}
|@math{15 \div 3 \times (28 - 22)}
|@math{15 \div (4 - 1) \times 2}
|===

}

@QandA{
@Q{What else did you notice and wonder? __Answers will vary!__ }
@A{Students may have noticed that the first 4 expressions evaluate to the same thing.}
@A{Students may have noticed that included circles can be on the left or on the right.}

@Q{How would we draw the Circle of Evaluation for the expression @math{10 - (5 + 3)}?}
@A{@show{(coe '(- 10 (+ 5 3)))}}
}
}

```

**Comment by schanzer** [comment]
@flannery-denny I can help with _some_ of this.

- COEs can _totally_ be inline...but not until we find a way to make asciidoc use <div> tags instead of <p> tags OR we get rid of exporting COEs to GDrive
- @strategy boxes _can_ contain tables, but they needed to use nested syntax (! instead of |)
- @ds26gte will have to weigh in on tables-inside-QandA, since I believe that's a parsing issue

**Comment by flannery-denny** [comment]
@ds26gte Possible make it so that I can include tables inside of `@QandA` ? You'll see where my build is breaking in the second second of the lesson plan in that last commit. Thanks!

**Comment by flannery-denny** [comment]
@schanzer I think this lesson is about ready for your review. And there is also zero rush!

There are two items remaining on my radar.
1)  tables inside `QandA` - ok to make a separate issue for that?
2) you requested that we remove negatives from required pages. I'm happy to do that, but before I do, I wanted to ask, how do you feel about leaving negatives in situations where students don't have to write them. For example, on a completing the code page, where the negatives have been handled by us and the student is only engaging with the positive numbers? Or on a printable set of cards to match.

**Comment by schanzer** [comment]
@flannery-denny Here are my comments on the lesson:

### Order of Operations

- "Circles of Evaluation - Notice and Wonder" seems overly complex.  Are you really expecting a notice and wonder for _each_ circle? Are they going to be unique? If so, there's not enough space to write them here -- and I worry that by asking for so many we're biasing kids to give really shallow answers. By not allowing students to notice and wonder about all of the circles together, this page makes it difficult for students to make broad observations like "every circle seems to only have one operation on top." I'd recommend _either_ (1) reducing the number of circles on this page by half and giving them more space to write or (2) show them two circles and their arithmetic equivalents, and give them more space to write.
- "The rules for drawing Circles of Evaluation are simple..." this should not use `@lesson-instruction`, since there's no direct action for students to take.
- I'm confused by the QandA box that follows. Shouldn't we start by asking students what they noticed and wondered? I would also ask them to draw the circle for `10 - (5 + 3)` _before_ asking them to translate all the ones on the page.
- In the Investigate section, Matching Expressions to Diagrams should be listed before "Complete the Circles...". In fact, this page feels like it might warrant being included in the workbook, to be completed immediately following the QandA box. It's very fast, and would help solidify students' understanding of what they just did _before_ doing the harder activities that follow.
- "Expressions -> Circles of Evaluation" You told me you were going to use the same expressions that show up later in "Expressions -> Circles of Evaluation -> Code 1". Why the change? I think having the continuity here is really valuable.
- In the Synthesize section, you'll need to explain what "inside-out" and "outside-in" mean. In fact, I would recommend bringing this up long before the synthesize section, so that teachers can introduce this difference and have kids try some of the Investigate activities using different approaches. This would also allow for a deeper synthesize question: _which do you prefer, and why?_

### From Circles of Evaluation to Code

- The first `@lesson-instruction` doesn't tell students to take any action, so the directive should be removed.
- I think it would be worth adding a strategy box to talk about putting up expressions around the room, and having students work in teams to translate into circles and then code. 

### Testing Our Your Code

- I LOVE "Notice and Wonder - More than +, - , ÷, ×"!!!
-

**Comment by flannery-denny** [comment]
@schanzer I think this lesson is radically improved. we could add a challenge page with brackets and curly brackets that didn't involve square roots, so it didn't hinge on getting to how pyret handles functions first. lmk if you think it's worth it or close as is.

**Comment by schanzer** [comment]
I think it looks fantastic. OK to close. Nice work!
